app/controllers/microposts_controller.rb

```diff
@@ -5,27 +5,60 @@ class MicropostsController < ApplicationController
   before_action :logged_in_user, only: %i[create destroy]
   before_action :correct_user, only: :destroy
   def create
-    @micropost = current_user.microposts.build(micropost_params)
+    if params[:micropost][:micropost_id]
+      micropost = Micropost.find_by(id: params[:micropost][:micropost_id])
+      @micropost = micropost.microposts.build(micropost_params)
+    else
+      @micropost = current_user.microposts.build(micropost_params)
+    end
     @micropost.image.attach(params[:micropost][:image])
+
     if @micropost.save
-      flash[:success] = 'Micropost created!'
-      redirect_to root_url
+      handle_micropost_save
+      handle_comment_save(@micropost)
     else
-      @feed_items = current_user.feed.paginate(page: params[:page])
+      @feed_items = current_user.feed.without_micropost_id.paginate(page: params[:page])
       render 'static_pages/home'
     end
   end
 
   def destroy
-    @micropost.destroy
+    @micropost.destroy!
```

Comment:
When does an error occur when destroying?
Also, I haven't seen you handle any errors.